### PR TITLE
#757 Validate input param for API and add UT for Controller

### PR DESCRIPTION
**Pull request by nashtech-hieunguyenchi5**
(cherry picked from commit 2a95f45f1a3d9105ce508149f9e64775d5af193e)

### CODE REVIEW

File: inventory/pom.xml [hunk 1]
```diff
@@ -52,6 +52,11 @@
             <groupId>org.liquibase</groupId>
             <artifactId>liquibase-core</artifactId>
         </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <scope>test</scope>
+        </dependency>
     </dependencies>
```
Comment (khanhtrand) [inline inventory/pom.xml:60]: Why do we need this dependency ? I think we already have spring-boot-starter-test in parent pom.xml

File: promotion/pom.xml [hunk 2]
```diff
@@ -52,6 +52,11 @@
             <artifactId>h2</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-test-autoconfigure</artifactId>
+            <scope>test</scope>
+        </dependency>
```
Comment (khanhtrand) [inline promotion/pom.xml:59]: Same as the above comment. I think we don't need this

File: inventory/src/main/java/com/yas/inventory/viewmodel/warehouse/WarehousePostVm.java [hunk 3]
```diff
@@ -1,8 +1,20 @@
...
+import jakarta.validation.constraints.NotNull;
+import lombok.Builder;
+import org.hibernate.validator.constraints.Length;
 
-public record WarehousePostVm(String id, @NotBlank String name, String contactName, String phone, String addressLine1, String addressLine2, String city, String zipCode,
-                              Long districtId, Long stateOrProvinceId, Long countryId) {
-
+@Builder
+public record WarehousePostVm(String id,
+                              @NotBlank @Length(max=450) String name,
```
Comment (khanhtrand) [inline inventory/src/main/java/com/yas/inventory/viewmodel/warehouse/WarehousePostVm.java:10]: Can we use jakarta.validation.constraints.Size for consistency ?
We should define min value to remove the `@NotBlank.` For example: `@Size(min = 1, max = 450)`

File: inventory/pom.xml (hunk 1 above)
Comment (nashtech-hieunguyenchi5) [reply inventory/pom.xml]: My mistake, I removed this dependency

File: promotion/pom.xml (hunk 2 above)
Comment (nashtech-hieunguyenchi5) [reply promotion/pom.xml]: My mistake, I removed this dependency

File: inventory/src/main/java/com/yas/inventory/viewmodel/warehouse/WarehousePostVm.java (hunk 3 above)
Comment (nashtech-hieunguyenchi5) [reply inventory/src/main/java/com/yas/inventory/viewmodel/warehouse/WarehousePostVm.java]: I agree with you about that.
I changed @Length to @Size

**Review by khanhtrand — APPROVED** [review]

**Review by thiennn — APPROVED** [review]

**Pull request merged**